x-pack/plugins/uptime/server/lib/requests/search/refine_potential_matches.ts

```diff
@@ -19,80 +18,83 @@ import { MonitorGroups, MonitorLocCheckGroup } from './fetch_page';
 export const refinePotentialMatches = async (
   queryContext: QueryContext,
   potentialMatchMonitorIDs: string[]
-): Promise<MonitorGroups[]> => {
+): Promise<MonitorSummary[]> => {
   if (potentialMatchMonitorIDs.length === 0) {
     return [];
   }
 
   const queryResult = await query(queryContext, potentialMatchMonitorIDs);
-  const recentGroupsMatchingStatus = await fullyMatchingIds(queryResult, queryContext.statusFilter);
-
-  // Return the monitor groups filtering out potential matches that weren't current
-  const matches: MonitorGroups[] = potentialMatchMonitorIDs
-    .map((id: string) => {
-      return { id, groups: recentGroupsMatchingStatus.get(id) || [] };
-    })
-    .filter((mrg) => mrg.groups.length > 0);
-
-  // Sort matches by ID
-  matches.sort((a: MonitorGroups, b: MonitorGroups) => {
-    return a.id === b.id ? 0 : a.id > b.id ? 1 : -1;
-  });
-
-  if (queryContext.pagination.cursorDirection === CursorDirection.BEFORE) {
-    matches.reverse();
-  }
-  return matches;
+  return await fullyMatchingIds(queryResult, queryContext.statusFilter);
```

Comment:
```suggestion
  return fullyMatchingIds(queryResult, queryContext.statusFilter);
```

Reply:
Yes, I'm aware that we can rely on the promise returned here. I think it's still good practice to always await. The reason being if this is ever copy/pasted it can prevent bugs. We've had some issues with await accidentally not being called in a similar scenario. Given that the perfect impact here approaches zero, I'd prefer to err on the side of over `await`ing.